packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -88,6 +106,10 @@ export default util.createRule({
         const leftType = getNodeType(node.left);
         const rightType = getNodeType(node.right);
 
+        if (option.allowAny && (leftType === 'any' || rightType === 'any')) {
```

Comment:
this is not quite correct.
This shouldn't override all validation in the rule, it should only allow `any` as a valid type.

I.e.
`any + boolean` shouldn't be allowed - because `boolean` isn't a valid type for this rule without the option.
However `any + (string | number | bigint)` should be allowed, as those 3 are all valid types.